source/deployment/deployContracts.ts

```diff
@@ -1,45 +1,46 @@
+import * as getPort from "get-port";
 import { SolidityContractCompiler } from "../libraries/CompileSolidity";
 import { ContractDeployer } from "../libraries/ContractDeployer";
 import { initializeTestRpcClientOptions } from "../libraries/HelperFunctions";
 import { RpcClient } from "../libraries/RpcClient";
+import { Connector } from '../libraries/Connector';
+import { Configuration } from '../libraries/Configuration';
 
 const CONTRACT_INPUT_DIR_PATH = path.join(__dirname, "../../source/contracts");
 const CONTRACT_OUTPUT_DIR_PATH = path.join(__dirname, "../../output/contracts");
 const COMPILED_CONTRACT_OUTPUT_FILE_NAME = "augurCore";
 
-export async function compileAndDeployContracts(): Promise<ContractDeployer> {
+export async function compileAndDeployContracts(configuration: Configuration, connector: Connector): Promise<ContractDeployer> {
     // Compile contracts to a single output file
     const solidityContractCompiler = new SolidityContractCompiler(CONTRACT_INPUT_DIR_PATH, CONTRACT_OUTPUT_DIR_PATH, COMPILED_CONTRACT_OUTPUT_FILE_NAME);
     const compilerOutput = await solidityContractCompiler.compileContracts();
 
-    const ethjsHttpProviderHost = (typeof process.env.ETHEREUM_HOST === "undefined") ? "localhost" : process.env.ETHEREUM_HOST;
-    const ethjsHttpProviderPort = (typeof process.env.ETHEREUM_PORT === "undefined") ? await getPort() : parseInt(process.env.ETHEREUM_PORT || "0");
-    const gasPrice = ((typeof process.env.ETHEREUM_GAS_PRICE_IN_NANOETH === "undefined") ? 20 : parseInt(process.env.ETHEREUM_GAS_PRICE_IN_NANOETH || "20")) * 10**9;
     // If no Ethereum host has been specified, use TestRPC.
     if (typeof process.env.ETHEREUM_HOST === "undefined") {
-        const testRpcOptions = await initializeTestRpcClientOptions(ethjsHttpProviderPort, ["Augur0"]);
+        const testRpcOptions = initializeTestRpcClientOptions(configuration.httpProviderPort, configuration.privateKey);
         const rpcClient = new RpcClient();
-        await rpcClient.listen(ethjsHttpProviderPort, testRpcOptions);
+        await rpcClient.listen(configuration.httpProviderPort, testRpcOptions);
     }
-    const ethjsHttpProvider = new EthjsHttpProvider(`http://${ethjsHttpProviderHost}:${ethjsHttpProviderPort}`);
-    const ethjsQuery = new EthjsQuery(ethjsHttpProvider);
-    const contractDeployer = new ContractDeployer(ethjsQuery, compilerOutput, gasPrice);
+    const ethjsQuery = await connector.waitUntilConnected();
+    const contractDeployer = new ContractDeployer(configuration, connector, compilerOutput);
     await contractDeployer.deploy();
 
     return contractDeployer;
 }
 
 // If this script is not being imported by another module (i.e., it is being run independently via the command line)
-if (!module.parent) {
+async function runStandalone() {
     require('source-map-support').install();
-    compileAndDeployContracts().then(() => {
+    const configuration = new Configuration(await getPort());
```

Comment:
I would move the `await getPort()` into Configuration so it can be co-located with the logic around undefined `ETHEREUM_HOST` && using TestRPC.

It's confusing to use `getPort` in a case where the seeming use-case is connecting to an external host -- not spinning up a server yourself.

Reply:
I did that originally, but `getPort` is async and you can't have an async constructor in TS, which would mean two-step construction.  What I can do though, now that I have slept and can think again, is add a `ConfigurationFactory()` method that is async that gets the port and returns a newly constructed `Configuration` object.

Reply:
Made the changes to have a factory function.  Check them out and let me know if that makes you feel better.